pkg/gatewaysecret/handler.go

```diff
@@ -26,48 +26,57 @@ const (
 	istioNamespace           = "istio-system"
 )
 
-var errCouldNotGetLastModifiedAt = errors.New("getting lastModifiedAt time failed")
+var (
+	errCouldNotGetLastModifiedAt = errors.New("getting lastModifiedAt time failed")
+	errExpectedOneRootCASecret   = errors.New("expected exactly one root CA secret")
+)
 
-type GatewaySecretHandler struct {
-	kcpClient client.Client
+type Handler struct {
+	kcpClient    client.Client
+	kcpClientset *kubernetes.Clientset
```

Comment:
I think it make more sense only use `Clientset` in this case, then this package is decoupled with `controller-runtime`

Reply:
actually you can design another struct to handle `GetRootCACertificate`, `RequiresUpdate`, then you can even decouple certmanger